synapse/app/homeserver.py

```diff
@@ -275,6 +275,7 @@ def setup():
     hs.get_pusherpool().start()
     hs.get_state_handler().start_caching()
     hs.get_datastore().start_profiling()
+    hs.get_replication_layer().start_pdu_fail_cache()
```

Comment:
Might nice to call this something other than "fail_cache" if it caches more than just failures.